.travis.yml

```diff
@@ -1,42 +1,72 @@
-language: python
+# use miniconda for python, so don't specify "python", which doesn't
+# work on travis osx anyway
+language: generic
+os:
+  - linux
+  - osx
+  # TODO: windows CI (not currently supported by travis)
+
+env:
+  global:
+    - USE_CONDA=true
+  matrix:
+    - QPYVER=2.7
+    - QPYVER=3.4
+    - QPYVER=3.5
+    - QPYVER=3.6
 
 matrix:
+  allow_failures:
+    # no wheel support, so pyarrow installation fails
+    - env: TRAVIS_OS_NAME=osx, QPYVER=3.4
+  fast_finish: true
   include:
     - os: linux
+      language: python
+      python: "2.7"
+      env: USE_CONDA=false
       sudo: required
```

Comment:
I suspect that `sudo` is not required anymore? Though I guess we might need it later.